pkg/auth/providers/common/usermanager.go

```diff
@@ -238,21 +240,38 @@ func (m *userManager) EnsureUser(principalName, displayName string) (*v3.User, e
 		return nil, err
 	}
 
-	logrus.Infof("Creating globalRoleBinding for %v", created.Name)
-	_, err = m.globalRoleBindings.Create(&v3.GlobalRoleBinding{
-		ObjectMeta: v1.ObjectMeta{
-			GenerateName: "globalrolebinding-",
-		},
-		UserName:       created.Name,
-		GlobalRoleName: "user",
-	})
+	logrus.Infof("Creating globalRoleBindings for %v", created.Name)
+	err = m.createNewUsersBindings(created.Name)
 	if err != nil {
 		return nil, err
 	}
 
 	return created, nil
 }
 
+func (m *userManager) createNewUsersBindings(name string) error {
+	roles, err := m.globalRoleLister.List("", labels.NewSelector())
+	if err != nil {
+		return err
+	}
+	for _, gr := range roles {
+		if gr.NewUserDefault {
+			_, err := m.globalRoleBindings.Create(&v3.GlobalRoleBinding{
+				ObjectMeta: v1.ObjectMeta{
+					GenerateName: "globalrolebinding-",
+				},
+				UserName:       name,
+				GlobalRoleName: gr.Name,
+			})
+
+			if err != nil && !errors2.IsAlreadyExists(err) {
```

Comment:
your're never going to get this error bc we are asking k8s to generate the name. But, we do have a different error scenario we should think through:
1. controller launches EnsureUser for userX. There are three default global roles A, B, C
2. userx is created
3. The GRB for role A is created successfully
4. The GRB for role B FAILS becuase of something random, like a network failure
5. error is returned to controller which then gives up and returns the error
6. framework reschedules the resource that kicked off this flow. userx already exists, thats fine. But then, ANOTHER GRB for role A is created. now we have ttwo GRBs for role A